src/imfilter_cuda.jl

```diff
@@ -0,0 +1,46 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+function imfilter_device(device::CUDALibs{R},
+                         img::AbstractArray{T,N},
+                         kern::AbstractArray{K,N}) where {R, T<:Real,K<:Real,N}
+
+    img_gpu_fft = nothing
+    kern_gpu_fft = nothing
+
+    size_all = Tuple(map(t->t[1]+t[2]-1,zip(size(img),size(kern))))
```

Comment:
This code is not N-dimensional, I think it is assuming 2D images?

Reply:
The code is N-dimensional. It was tested with one, two and three dimensional and the resultls were compared with the results of JuliaImages' imfilter. 
This code takes the sizes of the img and kern (N-tuples) and create another N-tuple by adding corresponding elements of the original tuples and subtracting one. Please, see the meaning of the zip function.